### ISSUE TITLE
postfix-rbl.conf regex for "454 4.7.1" should be "554 5.7.1" for default postfix reject_rbl_client

### ISSUE THREAD

**Issue by dbrucker** [fail2ban/fail2ban#1634]
The default /etc/fail2ban/filter.d/postfix-rbl.conf is looking for the wrong regex/code according to what postfix logs. /etc/fail2ban/filter.d/postfix-rbl.conf is looking for "...454 4.7.1...". Enabling "reject_rbl_client" in /etc/postfix/main.cf logs offenders as "...554 5.7.1..." which misses the fail2ban rule.
### Environment:
# in a default CentOS 7 install inside a VPS
```
[root@server filter.d]# rpm -q postfix
postfix-2.10.1-6.el7.x86_64
[root@server filter.d]# rpm -q fail2ban
fail2ban-0.9.5-3.el7.noarch

[root@server filter.d]# uname -a
Linux server.host 3.10.0-229.1.2.el7.x86_64 #1 SMP Fri Mar 27 03:04:26 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```
###
```
# bottom of (modified) /etc/postfix/main.cf
# TLS configuration
smtpd_use_tls = yes
smtpd_tls_cert_file = /etc/pki/tls/certs/server.crt
smtpd_tls_key_file = /etc/pki/tls/certs/server.key
smtpd_tls_session_cache_database = btree:/etc/postfix/smtpd_scache

# HELO restrictions
smtpd_helo_required = yes
smtpd_helo_restrictions =
  permit_mynetworks,
  reject_non_fqdn_helo_hostname,
  reject_invalid_helo_hostname,
  permit

smtpd_sender_restrictions =
  permit_mynetworks,
  reject_non_fqdn_sender,
  reject_unknown_sender_domain,
  permit

smtpd_recipient_restrictions =
  reject_unauth_pipelining,
  reject_non_fqdn_recipient,
  reject_unknown_recipient_domain,
  permit_mynetworks,
  reject_unauth_destination,
  reject_unknown_reverse_client_hostname,
  reject_rbl_client zen.spamhaus.org,
  reject_rbl_client bl.spamcop.net,
  reject_rbl_client b.barracudacentral.org,
  reject_rbl_client dnsbl.sorbs.net,
  check_policy_service unix:postgrey/socket,
# check_policy_service unix:private/policy,
  permit

# for OpenDKIM
smtpd_milters = inet:127.0.0.1:8891
non_smtpd_milters = $smtpd_milters
milter_default_action = accept
```

### The issue:

The default /etc/fail2ban/filter.d/postfix-rbl.conf is looking for the wrong regex/code according to what postfix logs. /etc/fail2ban/filter.d/postfix-rbl.conf is looking for "...454 4.7.1...". Enabling "reject_rbl_client" in /etc/postfix/main.cf logs offenders as "...554 5.7.1..." which misses the fail2ban rule.

#### Steps to reproduce
1. install postfix and fail2ban/postfix-rbl with at least one instance of "reject_rbl_client xxx" on incoming mail
2. wait for spammers to send mail from RBL'd ip's
3. fail2ban-regex /var/log/maillog /etc/fail2ban/filter.d/postfix-rbl.conf shows all lines as "missed"

#### Expected behavior
```
[root@server filter.d]# fail2ban-regex /var/log/maillog /etc/fail2ban/filter.d/postfix-rbl.conf 

Running tests
=============

Use   failregex filter file : postfix-rbl, basedir: /etc/fail2ban
Use         log file : /var/log/maillog
Use         encoding : ANSI_X3.4-1968


Results
=======

Failregex: 2 total
|-  #) [# of hits] regular expression
|   1) [2] ^(?:\[\])?\s*(?:<[^.]+\.[^.]+>\s+)?(?:\S+\s+)?(?:kernel: \[ *\d+\.\d+\]\s+)?(?:@vserver_\S+\s+)?(?:(?:(?:\[\d+\])?:\s+[\[\(]?postfix(-\w+)?/smtpd(?:\(\S+\))?[\]\)]?:?|[\[\(]?postfix(-\w+)?/smtpd(?:\(\S+\))?[\]\)]?:?(?:\[\d+\])?:?)\s+)?(?:\[ID \d+ \S+\]\s+)?NOQUEUE: reject: RCPT from \S+\[<HOST>\]: 554 5\.7\.1 Service unavailable; Client host \[\S+\] blocked using .* from=<\S*> to=<\S+> proto=ESMTP helo=<\S*>$
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [5768] (?:DAY )?MON Day 24hour:Minute:Second(?:\.Microseconds)?(?: Year)?
`-

Lines: 5768 lines, 0 ignored, 2 matched, 5766 missed
[processed in 0.51 sec]
```

#### Observed behavior
```
[root@server filter.d]# fail2ban-regex /var/log/maillog /etc/fail2ban/filter.d/postfix-rbl.conf 

Running tests
=============

Use   failregex filter file : postfix-rbl, basedir: /etc/fail2ban
Use         log file : /var/log/maillog
Use         encoding : ANSI_X3.4-1968


Results
=======

Failregex: 0 total

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [5791] (?:DAY )?MON Day 24hour:Minute:Second(?:\.Microseconds)?(?: Year)?
`-

Lines: 5791 lines, 0 ignored, 0 matched, 5791 missed
[processed in 0.49 sec]

Missed line(s): too many to print.  Use --print-all-missed to print all 5791 lines
```

#### Any additional information
This is inside a bone-stock CentOS 7 install

### Configuration, dump and another helpful excerpts
```
[root@server filter.d]# postconf -d | grep reject_
access_map_reject_code = 554
invalid_hostname_reject_code = 501
maps_rbl_reject_code = 554
message_reject_characters =
multi_recipient_bounce_reject_code = 550
non_fqdn_reject_code = 504
plaintext_reject_code = 450
postscreen_reject_footer = $smtpd_reject_footer
reject_code = 554
reject_tempfail_action = defer_if_permit
relay_domains_reject_code = 554
smtpd_reject_footer =
smtpd_reject_unlisted_recipient = yes
smtpd_reject_unlisted_sender = no
unknown_address_reject_code = 450
unknown_address_tempfail_action = $reject_tempfail_action
unknown_client_reject_code = 450
unknown_helo_hostname_tempfail_action = $reject_tempfail_action
unknown_hostname_reject_code = 450
unknown_local_recipient_reject_code = 550
unknown_relay_recipient_reject_code = 550
unknown_virtual_alias_reject_code = 550
unknown_virtual_mailbox_reject_code = 550
unverified_recipient_reject_code = 450
unverified_recipient_reject_reason =
unverified_recipient_tempfail_action = $reject_tempfail_action
unverified_sender_reject_code = 450
unverified_sender_reject_reason =
unverified_sender_tempfail_action = $reject_tempfail_action
```

#### Any customizations done to /etc/fail2ban/ configuration
```
```

#### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

```
```

#### Relevant lines from monitored log files in question:
```
[root@server1 ~]# grep "554 5.7.1" /var/log/maillog
Nov 30 08:59:29 server1 postfix/smtpd[16517]: NOQUEUE: reject: RCPT from unknown[181.174.189.209]: 554 5.7.1 Service unavailable; Client host [181.174.189.209] blocked using zen.spamhaus.org; https://www.spamhaus.org/sbl/query/SBLCSS; from=<9925-8924746559-861-sales=host.com@mail.desirre.us> to=<sales@host.com> proto=ESMTP helo=<host.desirre.us>
Dec  3 03:23:31 server1 postfix/smtpd[12861]: NOQUEUE: reject: RCPT from unknown[181.174.189.197]: 554 5.7.1 Service unavailable; Client host [181.174.189.197] blocked using zen.spamhaus.org; https://www.spamhaus.org/sbl/query/SBLCSS; from=<10005-1377585512-889-sales=host.com@mail.navvjohearsystm.us> to=<sales@host.com> proto=ESMTP helo=<sioux.navvjohearsystm.us>
```

**Comment by sebres** [comment]
@leeclemens What do you think about this?

**Comment by dbrucker** [comment]
Does it make more sense to just add a second rule to cover the second case? I'm not sure if this is just a preference or a policy of the project. But it would make it "just work" for either reject code. (And thank you to whomever properly formatted my original report.)

**Comment by bmatthewshea** [comment]
Yes - Had to update my postfix-rbl filter to look for "554 - 5.7.1" as well.
Using Ubuntu 16.04.1
- Postfix (Ubuntu repo pkg) - Vers 3.1.0-3 (using default error codes for logging)
- Custom fail2ban install (newest)

Sample postfix log:
Jan 16 21:42:39 myhostname postfix/smtpd[2461]: NOQUEUE: reject: RCPT from h2232618.xxxxxxxxxxxx.xxx[XX.XXX.143.25]: **554 5.7.1** Service unavailable; Client host [XX.XXX.143.25] blocked using zen.spamhaus.org; https://www.spamhaus.org/query/ip/xx.xxx.143.25; from=<angela@xxxxxx.xxxxxxxxxxxx.xxx> to=<webmaster@xxxxxxxxx.xxx> proto=ESMTP helo=<server.xxxxxxxxxxx.xxx>

**Comment by sebres** [comment]
fixed within #1686 (and cherry-picked to v.0.9 also)
